actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/dao/impl/UserSelfDeclarationDaoImpl.java

```diff
@@ -0,0 +1,97 @@
+package org.sunbird.user.dao.impl;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.collections4.MapUtils;
+import org.sunbird.cassandra.CassandraOperation;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.request.RequestContext;
+import org.sunbird.helper.ServiceFactory;
+import org.sunbird.models.user.UserDeclareEntity;
+import org.sunbird.user.dao.UserSelfDeclarationDao;
+
+public class UserSelfDeclarationDaoImpl implements UserSelfDeclarationDao {
+  private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+  private ObjectMapper mapper = new ObjectMapper();
+
+  private static UserSelfDeclarationDao userSelfDeclarationDao = null;
+
+  public static UserSelfDeclarationDao getInstance() {
+    if (userSelfDeclarationDao == null) {
+      userSelfDeclarationDao = new UserSelfDeclarationDaoImpl();
+    }
+    return userSelfDeclarationDao;
+  }
+
+  public void insertSelfDeclaredFields(Map<String, Object> extIdMap, RequestContext context) {
+
+    cassandraOperation.insertRecord(
+        JsonKey.SUNBIRD, JsonKey.USER_DECLARATION_DB, extIdMap, context);
+  }
+
+  public List<Map<String, Object>> getUserSelfDeclaredFields(
+      UserDeclareEntity userDeclareEntity, RequestContext context) {
+    List<Map<String, Object>> dbResExternalIds = new ArrayList<>();
```

Comment:
Please add loggers for the entry and exit.

Reply:
Entry and exit logs has to be in API level,  so another ticket will take care of this for all APIs